webapp/loggedin/modules/home/surveyList/surveyListView.js

```diff
@@ -1,42 +1,54 @@
 import './surveyListView.scss'
 
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { connect } from 'react-redux'
 import * as R from 'ramda'
 
-import SurveyListTable from './surveyListTable'
-
 import Survey from '../../../../../common/survey/survey'
 
+import SurveyListTable from './surveyListTable'
+
 import * as SurveyListState from './surveyListState'
 import * as SurveyState from '../../../../survey/surveyState'
+import { appModuleUri, homeModules } from '../../../appModules'
 
 import { fetchSurveys } from './actions'
 import { setActiveSurvey } from '../../../../survey/actions'
 
-class SurveyListView extends React.Component {
-
-  componentDidMount () {
-    this.props.fetchSurveys()
-  }
-
-  render () {
-    const { surveyInfo, surveys, setActiveSurvey } = this.props
-
-    return R.isEmpty(surveys)
-      ? null : (
-        <SurveyListTable
-          surveys={surveys}
-          surveyInfo={surveyInfo}
-          setActiveSurvey={setActiveSurvey}
-        />
-      )
-  }
+const SurveyListView = props => {
+
+  const {
+    surveyInfo, surveys, history,
+    setActiveSurvey, fetchSurveys
+  } = props
+
+  const [fetched, setFeched] = useState(false)
+
+  //onMount fetch surveys
+  useEffect(() => {
+    fetchSurveys()
+    setFeched(true)
+  }, [])
+
+  const surveysLength = R.length(surveys)
+  useEffect(() => {
+    // redirect to create survey when (after fetching surveys) there are no surveys
+    if (fetched === true && surveysLength === 0) {
+      history.push(appModuleUri(homeModules.surveyNew))
+    }
+  }, [surveysLength])
```

Comment:
Should it be [surveysLength, fetched]?

Reply:
It does't matter. at that point is always true, because the surveys have been fetched, therefore the surveysLength has changed.